src/ucs/sys/topo/base/topo.h

```diff
@@ -72,6 +72,10 @@ typedef ucs_status_t
 extern ucs_list_link_t ucs_sys_topo_providers_list;
 
 
+/* Selected topo provider */
+extern struct ucs_sys_topo_provider *ucs_sys_topo_provider;
```

Comment:
Ok, I think it's better to have a reset API rather then extending `ucs_sys_topo_get_provider` with reset flag, because the latter would require one extra check in the production flow